cpp/src/io/csv/reader_impl.cu

```diff
@@ -30,7 +30,7 @@
 #include "legacy/datetime_parser.cuh"
```

Comment:
datetime_parser.cuh doesn't appear to use any legacy functionality so it can be copied into a new (non-legacy) path.  The copy should be modified to not include (legacy) `cudf.h`.